keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -218,7 +233,16 @@ interface ResourceRepository : PeriodicallyCheckedRepository<Resource<out Resour
   fun Resource<*>.toResourceSummary() =
     ResourceSummary(
       resource = this,
-      status = getStatus(id) // todo eb: this will become expensive
+      status = getStatus(id), // todo eb: this will become expensive
+      artifact = if (spec is ArtifactProvider) {
```

Comment:
Here, upon retrieval of the resource from the database, we will have filled in the artifact type and name based on the association in the `resource_artifact` table, and so the `spec` will qualify as an `ArtifactProvider` with an artifact name and type we can add to the summary.